src/sentry/api/endpoints/organization_events_trace.py

```diff
@@ -134,49 +150,100 @@ def get(self, request, organization, trace_id):
-        for item in result["data"]:
-            if not is_root(item):
-                parent_map[item["trace.parent_span"]].append(item)
-
         # Temporarily feature flagging this out, since errors will impact performance
         if not features.has("organizations:trace-view-summary", organization, actor=request.user):
-            error_map = []
+            errors = []
         else:
-            error_map = self.get_error_map(organization, trace_id, params)
+            current_transaction = find_event(result["data"], lambda t: t["id"] == event_id)
+            errors = self.get_errors(organization, trace_id, params, current_transaction, event_id)
 
         return Response(
-            self.serialize(
-                parent_map, error_map, root, warning_extra, params, snuba_event, event_id, detailed
-            )
+            self.serialize(result["data"], errors, root, warning_extra, event_id, detailed)
         )
 
 
 class OrganizationEventsTraceLightEndpoint(OrganizationEventsTraceEndpointBase):
-    def get_error_map(self, *args, **kwargs):
-        """We don't get the error map for the light view
+    def get_current_event(self, transactions, errors, event_id):
+        """Given an event_id return the related transaction event
 
-        This is because we only get spans for the root + current event, which means we could only create an error
-        to transaction association for up to two events.
+        The event_id could be for an error, since we show the quick-trace
+        for both event types We occasionally have to get the nodestore data
+        so this function returns the nodestore event as well so that we're
+        doing that in one location.
         """
-        return {}
+        snuba_event = find_event(transactions, lambda item: item["id"] == event_id)
+        if snuba_event is not None:
+            return snuba_event, eventstore.get_event_by_id(
+                snuba_event["project.id"], snuba_event["id"]
+            )
+
+        # The event couldn't be found, it might be an error
+        error_event = find_event(errors, lambda item: item["id"] == event_id)
+        # Alright so we're looking at an error, time to see if we can find its transaction
+        if error_event:
+            # Unfortunately the only association from an event back to its transaction is name & span_id
+            # First maybe we got lucky and the error happened on the transaction's "span"
+            snuba_event = find_event(
+                transactions, lambda item: item["trace.span"] == error_event["trace.span"]
+            )
+            if snuba_event:
+                return snuba_event, eventstore.get_event_by_id(
+                    snuba_event["project.id"], snuba_event["id"]
+                )
+            # We didn't get lucky, time to talk to nodestore...
+            for transaction in filter(
+                lambda t: t["transaction"] == error_event["transaction"], transactions
+            ):
```

Comment:
Might be a personal preference, but I don't like `filter` in `python` where it's unnecessary.

```suggestion
            for transaction in transactions:
                if transaction["transaction"] != error_event["transaction"]:
                    continue
```

Reply:
Hmm I personally don't have a preference either way TBH, any specific reason to avoid filter? At least with simple conditions like this they're nearly identical in perf & readability to me

Reply:
My argument against `filter` is that these simple conditions evolve over time. We have a tendency to not change more than is necessary so this turns into a giant lambda expression which I would argue is a lot less legible compared to simple if statements. But again, this is just my preference.